src/services/ui/src/utils/Poller/DataPoller.ts

```diff
@@ -0,0 +1,45 @@
+export type Options<TFetcherReturn> = {
+  fetcher: () => TFetcherReturn;
+  onPoll: (data: Awaited<TFetcherReturn>) => boolean;
+  interval: number;
+  pollAttempts: number;
+};
+
+export class DataPoller<TFetcherReturn> {
+  private options: Options<TFetcherReturn>;
+
+  constructor(options: Options<TFetcherReturn>) {
+    this.options = options;
+  }
+
+  async startPollingData() {
+    return new Promise<{
+      maxAttemptsReached: boolean;
+      correctStatusFound: boolean;
```

Comment:
status nomenclature should be generalized to indicate all checks in a check func pass, not just a status change -- distinction here is key since we update a few times before all the data is updated and present.